### DISCUSSION TITLE
Web sites behind proxy companion

### DISCUSSION THREAD

**Discussion by Elmit2015** [evertramos/nginx-proxy-automation#310] [Q&A]
I need some help to understand the entire system here.
I understand that the front-end is the proxy companion and behind are the the other web sites. As example is docker-wordpress-letsencrypt.

1. How do I setup 2 web sites?
Do I use a directory for each web site with a docker-compose.yml customized for each web site? That would make the .env clear for each site. Or do I use one docker-compose.yml for all web sites and add sections for each web site (services)?
What is the best directory structure for multiple web sites? I am thinking on backup, or moving to other hosts. 

2. I have my web sites now running with nginx. These are web sites with different domain names, for which I do have different nginx config files. Some of the sites use authentication and some have a connection to a trading bot or chat bot, where I listen for incoming information.

3. How is the link between the proxy companion and the web site. I assume with:
`networks:
   default:
      name: ${NETWORK}
`

**Comment by Elmit2015** [comment]
Can anybody help, please?

**Comment by evertramos** [comment]
I will post a video soon on that... but basically you can check the repo wordpress... it should be on folder/repo for each site, change settings in env file and service name in the docker-compose file.

**Comment by cwilby** [comment]
@Elmit2015 As I understand it:

**How do I setup 2 web sites?**

> Do I use a directory for each web site with a docker-compose.yml customized for each web site? 

Yes.  For a two-site setup, I have three folders each with docker-compose.yml.  Two for the websites, one for nginx-proxy (a fork of this repository in case I make changes)

> What is the best directory structure for multiple web sites? I am thinking on backup, or moving to other hosts.

As I use three separate folders/repositories, the directory structure looks something like this in production:

```
/home/foo
|__ website1.com
|__ website2.com
|__ nginx-proxy
```

> I have my web sites now running with nginx. These are web sites with different domain names, for which I do have different nginx config files. Some of the sites use authentication and some have a connection to a trading bot or chat bot, where I listen for incoming information.

Not a problem.  In website1.com and website2.com, set the `VIRTUAL_HOST` to match the domain name of the website.  If you happen to be running subdomains or multiple domains pointing to the same instance, I can't help you there.

> How is the link between the proxy companion and the web site?

`nginx-proxy` creates a separate docker network when you run `./start.sh`.  The name of this network is defined in `${NETWORK}` environment variable, and by default is named `webproxy`.  

So, in your website docker-compose, add the network and set it to external:

```yaml
version: "3"

services:
  app:
    image: nginx:latest
    environment:
      VIRTUAL_HOST: "website1.com"
    networks:
      - webproxy

networks:
  webproxy:
    external: true
```

Hope that helps, as a disclaimer I'm still learning about the project so may have a few points wrong.

**Comment by Darksoulsong** [comment]
@cwilby I'm having some issues on making more than one website to work. If you got the time, could you kindly give me some insights? Thanks a lot!
#174

**Comment by cwilby** [comment]
@Darksoulsong  sure, I’m traveling at the moment but can share more later.  I have a solution running several websites on a single docker host using this repo: https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion

**Comment by evertramos** [comment]
@cwilby this project uses the same companion you have mentioned just atuomate things for you, I mention that in the readme.

**Comment by datenheim** [comment]
Why not use a discussion for such? Not an issue, could be closed again :)
